conjure-java-jaxrs-client/src/test/java/com/palantir/conjure/java/client/jaxrs/TracerTest.java

```diff
@@ -60,26 +54,31 @@ public void before() {
     @Test
     public void testClientIsInstrumentedWithTracer() throws InterruptedException {
         server.enqueue(new MockResponse().setBody("\"server\""));
-        OpenSpan parentTrace = Tracer.startSpan("");
-        List<Map.Entry<SpanType, String>> observedSpans = Lists.newArrayList();
-        Tracer.subscribe(TracerTest.class.getName(),
-                span -> observedSpans.add(Maps.immutableEntry(span.type(), span.getOperation())));
 
+        String parentSpanId = Tracer.startSpan("").getSpanId();
         String traceId = Tracer.getTraceId();
+
+        List<String> spanIds = Lists.newArrayList();
+        List<Span> spans = Lists.newArrayList();
+        Tracer.subscribe(TracerTest.class.getName(), span -> {
+            spanIds.add(span.getSpanId());
+            spans.add(span);
+        });
+
         service.param("somevalue");
 
         Tracer.unsubscribe(TracerTest.class.getName());
-        assertThat(observedSpans, contains(
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: acquire-limiter-enqueue"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: acquire-limiter-run"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: execute-enqueue"),
-                Maps.immutableEntry(SpanType.CLIENT_OUTGOING, "OkHttp: GET /{param}"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: execute-run"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: dispatcher")));
+
+        assertThat(spans).containsExactly(
+                span(spans.get(0), traceId, spanIds.get(2), SpanType.LOCAL, "OkHttp: limiter-enqueue"),
+                span(spans.get(1), traceId, spanIds.get(2), SpanType.LOCAL, "OkHttp: limiter-run"),
+                span(spans.get(2), traceId, parentSpanId, SpanType.LOCAL, "OkHttp: enqueue"),
+                span(spans.get(3), traceId, spanIds.get(4), SpanType.CLIENT_OUTGOING, "OkHttp: GET /{param}"),
+                span(spans.get(4), traceId, spanIds.get(1), SpanType.LOCAL, "OkHttp: dispatcher"));
```

Comment:
It might also make sense to just get rid of the dispatcher instrumentation. It looks like that's what #1069 does.